### PR TITLE
Prepare for v0.2.0 release

### PR DESCRIPTION
**Pull request by tomhjp**
There have been a _lot_ of updates since the previous release so the binaries attached to that 0.1.0 release are very outdated. I'm planning to cut a 0.2.0 tag and attach some artifacts to the GitHub release in the same way, but I won't publish to releases.hashicorp.com as this is not a production-grade plugin.

* Update file structure to fit in with that expected by plugin release automation
* Update deps
* Strip away outdated Makefile targets
* Update version to 0.2.0 and implement the plugin version interface
* Use vault-workflows-common jira sync action

### CODE REVIEW

**Review by swenson — APPROVED** [review]
LGTM

File: .github/workflows/jira.yaml [hunk 1]
```diff
@@ -1,72 +1,17 @@
...
-      with:
-        issue: ${{ steps.search.outputs.issue }}
-        transition: "Pending Triage"
+    uses: hashicorp/vault-workflows-common/.github/workflows/jira.yaml@main
+    secrets:
+      JIRA_SYNC_BASE_URL: ${{ secrets.JIRA_SYNC_BASE_URL }}
+      JIRA_SYNC_USER_EMAIL: ${{ secrets.JIRA_SYNC_USER_EMAIL }}
+      JIRA_SYNC_API_TOKEN: ${{ secrets.JIRA_SYNC_API_TOKEN }}
+    with:
+      teams-array: '["ecosystem", "foundations-eco"]'
```
Comment (fairclothjm) [inline .github/workflows/jira.yaml:17]:
```suggestion
      teams-array: '["foundations-eco"]'
```

Comment (fairclothjm) [reply .github/workflows/jira.yaml:17]: or... are we still using the `ecosystem` label? If so then disregard

**Review by fairclothjm — APPROVED** [review]
LGTM! Now if we could only change the repo name to vault-**plugin**-auth-example 😄

File: .github/workflows/jira.yaml (hunk 1 above)
Comment (tomhjp) [reply .github/workflows/jira.yaml:17]: Oh good point, I don't think foundations should necessarily own this anyway, so I'll revert back to just the ecosystem label we were using previously.

**Comment by tomhjp** [comment]
Thanks!

**Pull request merged**